packages/astro/src/assets/utils/remoteProbe.ts

```diff
@@ -1,7 +1,7 @@
 import { lookup } from './vendor/image-size/lookup.js';
 import type { ISize } from './vendor/image-size/types/interface.ts';
 
-export async function probe(url: string): Promise<ISize> {
+export async function inferRemoteSize(url: string): Promise<ISize> {
```

Comment:
I don't really like that we're returning a type from `image-size` here, it means that `image-size` updates might require a minor or major for us in case it changes. Not to mention, it restricts us to `image-size`.

Maybe what should be exported is a wrapper of this function that return some other interface.

Reply:
What would you suggest? Just separate out the interface and duplicate the `image-size` types for now? Like a `RemoteISize` type?

Reply:
I presume I'd manually parse out the important bits so that it can be changed freely? Or just straight up return the same thing now, with the other type and allow for changes?

Reply:
Yeah, you can fish out the import bits. This is similar to how imageMetadata works here: https://github.com/withastro/astro/blob/main/packages/astro/src/assets/utils/metadata.ts

Reply:
I'm confused though, I do want `width`, `height`, and `type`, and you told that `orientation` sometimes shows up, these are all the results I want.

It'd duplicate the exact same type... I did remove the `orientation` property as i never saw it exposed, but the rest is the same...

Reply:
It's about not exposing to users a type we don't own unnecessarily. The content of the type is correct, it's just it's provenance that's annoying.

If `image-size` release an update tomorrow that change, I don't know, `type` to `format`, we can't update it in Astro without releasing a new major version of Astro.

Reply:
Ohhh, never mind, I didn't realize it wasn't fully owned by Astro.

So just probably use ImageMetadata, with the `Omit<>`...

Reply:
It's so annoying though, ImageMetadata uses format, ISize uses type 🙈